app/models/server/models/Users.js

```diff
@@ -204,6 +210,64 @@ export class Users extends Base {
 		return null;
 	}
 
+	// get next agent ignoring the ones reached the max amount of active chats
+	getAgentLoadByUsernames(usernames, customFilter) {
```

Comment:
Why do we need a list of usernames as parameter?
You are already filtering only online users with **livechat-agent** role.

Reply:
You're right on this. I used what the function already had, so I borrowed the list of usernames, but we can remove it :) (and pass a "department filter" as an extra param, since for the department usage, we don't want to query all users)